pahfit/model.py

```diff
@@ -241,47 +247,59 @@ def dust_continuum_guess(row):
         loop_over_non_fixed("dust_continuum", "tau", dust_continuum_guess)
 
         def line_fwhm_guess(row):
-            w = row["wavelength"][0]
-            if not instrument.within_segment(w, inst):
+            lam_line = row["wavelength"][0]
+            if not instrument.within_segment(lam_line, inst):
                 return 0
 
-            fwhm = instrument.fwhm(inst, w, as_bounded=True)[0][0]
+            fwhm = instrument.fwhm(inst, lam_line, as_bounded=True)[0][0]
             return fwhm
 
-        def amp_guess(row, fwhm):
-            w = row["wavelength"][0]
-            if not instrument.within_segment(w, inst):
+        def power_guess(row, fwhm):
+            # local integration for the lines
+            lam_line = row["wavelength"][0]
+            if not instrument.within_segment(lam_line, inst):
                 return 0
 
             factor = 1.5
-            wmin = w - factor * fwhm
-            wmax = w + factor * fwhm
-            lam_window = (lam > wmin) & (lam < wmax)
+            lam_min = lam_line - factor * fwhm
+            lam_max = lam_line + factor * fwhm
+            lam_window = (lam > lam_min) & (lam < lam_max)
             xpoints = lam[lam_window]
             ypoints = flux[lam_window]
             if np.count_nonzero(lam_window) >= 2:
                 # difference between flux in window and flux around it
-                power_guess = integrate.trapezoid(flux[lam_window], lam[lam_window])
+                Fnu_dlambda = integrate.trapezoid(flux[lam_window], lam[lam_window])
                 # subtract continuum estimate, but make sure we don't go negative
                 continuum = (ypoints[0] + ypoints[-1]) / 2 * (xpoints[-1] - xpoints[0])
-                if continuum < power_guess:
-                    power_guess -= continuum
+                if continuum < Fnu_dlambda:
+                    Fnu_dlambda -= continuum
             else:
-                power_guess = 0
-            return power_guess / fwhm
+                Fnu_dlambda = 0
+
+            # this is an unphysical power (Fnu * dlambda), but we
+            # convert to Fnu dnu = Fnu dnu/dlambda dlambda = Fnu c /
+            # lambda **2 dlambda
+            Fnu_dlambda *= units.intensity * units.wavelength
+            Fnu_dnu = Fnu_dlambda * constants.c / (lam_line * units.wavelength) ** 2
+            return Fnu_dnu.to(units.intensity_power).value
```

Comment:
Is this robust to future changes to accept flux in addition to intensity?  Ideally `Model` would take "whatever units" and DRT for the underlying `Fitter`'s.

Reply:
I think that we will have to implement some if/else logic regarding flux/intensity in various places. So far I have not really thought of a way around that.

For this piece of code in particular, the adaptation should be straightforward: just replace units.intensity with units.flux, and .to_units(intensity_power) by .to_units(flux_power).

This does mean we do need some way to flag and remember which type of flux quantity we're working with.

Reply:
> we do need some way to flag and remember which type of flux quantity we're working with 

In `Fitter` you mean?  One idea would be to standardize `Fitter` on intensity units, and convert flux units via some canonical assumed solid angle to `MJy/sr` (i.e. let `Model` handle those details).

Reply:
See #297 for a more detailed issue on this.  For now I think we just stick with supporting intensity units only.